File: 1037.valid-boomerang.py
#
# @lc app=leetcode id=1037 lang=python
#
# [1037] Valid Boomerang
#

# @lc code=start
class Solution(object):
    def isBoomerang(self, points):
        """
        :type points: List[List[int]]
        :rtype: bool
        """

        boom = True
        if points[0] == points[1] or points[1] == points[2] or points[0] == points[2]:
            boom = False
        else:
            if points[0][0] == points[1][0] == points[2][0]:
                boom = False
            elif points[0][1] == points[1][1] == points[2][1]:
                boom = False
            else:
                if (points[0][0] - points[1][0]) * (points[0][1] - points[2][1]) == (points[0][0] - points[2][0]) * (points[0][1] - points[1][1]):
                    boom = False
        return boom
# @lc code=end

File: 1037.valid-boomerang.js
/*
 * @lc app=leetcode id=1037 lang=javascript
 *
 * [1037] Valid Boomerang
 */

// @lc code=start
/**
 * @param {number[][]} points
 * @return {boolean}
 */
var isBoomerang = function(points) {
    const [x1, y1] = points[0];
    const [x2, y2] = points[1];

    if (x1 === x2 && y1 === y2) return false;

    const [x3, y3] = points[2];

    return (y2 - y1) * (x3 - x2) !== (y3 - y2) * (x2 - x1);
    
};
// @lc code=end
